corehq/apps/app_manager/dbaccessors.py

```diff
@@ -176,6 +176,35 @@ def get_built_app_ids_for_app_id(domain, app_id, version=None):
     return [result['id'] for result in results]
 
 
+def get_latest_built_app_ids_and_versions(domain, app_id=None):
+    """
+    Returns all the latest build app_ids and version in a dictionary:
+    {app_id: latest_version}
+    """
+    from .models import Application
+    key = [domain, app_id] if app_id else [domain]
+
+    results = Application.get_db().view(
+        'app_manager/saved_app',
+        startkey=key + [{}],
+        endkey=key,
+        descending=True,
+        reduce=False,
+        include_docs=False,
+    ).all()
+
+    latest_ids_and_versions = {}
+    for result in results:
+        app_id = result['key'][1]
+        version = result['key'][2]
+
+        # Since we have sorted, we know the first instance is the latest version
+        if app_id not in latest_ids_and_versions:
```

Comment:
it's basically a generalization. here's what gets returned from the view: [`domain`, `app_id`, `version`]. so we could get something like this with an app_id specified:
```
my-domain, 1234, 12
my-domain, 1234, 10
my-domain, 1234, 3
```
we loop over to just grab the first one. we don't need the loop if the app_id is specified. however if it's not specified you get something like this:
```
my-domain, 1234, 12
my-domain, 1234, 10
my-domain, 1234, 3
my-domain, 4567, 8
my-domain, 4567, 3
my-domain, 4567, 1
```
In which case we need to loop to get the resulting `{'1234': 12, '4567': 8 }`. can be a bit confusing do you think i should split it into two functions, one that takes an app_id and one that doesn't?